app/assets/javascripts/file_manager.js

```diff
@@ -0,0 +1,147 @@
+function rename(path) {
+    let new_name = prompt("Enter the new name:");
+    if (new_name !== null) {
+        let rel_path = decodeURIComponent(path.split("/file_manager/")[1]);
+        $.ajax({
+            url: "/file_manager/" + rel_path,
+            type: "PUT",
+            data: { new_name: new_name, relative_path: rel_path},
+            success: function(data) {
+                console.log(`Renamed: ${rel_path}`)
+                location.reload();
+            }
+        });
+    }
+}
+
+function deleteSelected(path) {
+    if (path !== null && path !== "") {
+        $.ajax({
+            url: path,
+            type: "DELETE",
+            success: function () {
+                console.log(`Deleted: ${path}`);
+                location.reload();
+            },
+            error: function (xhr, status, error) {
+                console.error(`Failed to delete ${path}: ${error}`);
+            }
+        });
+    }
+}
+
+function downloadSelected(path) {
+    $.ajax({
+        url: '/file_manager/download_tar/',
+        type: "POST",
+        data: {
+            path: path
+        },
+        success: function (data) {
+            let blob = new Blob([data], { type: 'application/x-tar' });
+            let url = URL.createObjectURL(blob);
+            let a = document.createElement('a');
+            a.href = url;
+            let parts = path.split("/")
+            a.download = parts[parts.length - 1]
+            a.style.display = 'none';
+            document.body.appendChild(a);
+            a.click();
+            document.body.removeChild(a);
+            URL.revokeObjectURL(url);
+            console.log(`Downloaded: ${data.filename}`);
+        },
+        error: function (xhr, status, error) {
+            console.error(`Failed to download ${path}: ${error}`);
+        }
+    });
+}
+
+function uploadFile(file, path, name) {
```

Comment:
I was able to break the upload file button by trying to upload a 23 GB file.  I'm not sure how relevant this is, but it might be necessary to check for the size of files being uploaded.

Reply:
Changed the limit size to 1GB.